src/App.tsx

```diff
@@ -1,26 +1,291 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { TodoItem } from './components/TodoItem';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [filter, setFilter] = useState<FilterType>(FilterType.All);
+  const [isLoading, setIsLoading] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [tempoTodo, setTempoTodo] = useState<Todo | null>(null);
+  const [IdToLoad, setIdToLoad] = useState<number[]>([]);
```

Comment:
```suggestion
  const [processingIds , setIdToLoad] = useState<number[]>([]);
```
do you only load?
i believe update and delete to so processingIds suits better